src/main/java/manga/MangaList.java

```diff
@@ -1,20 +1,40 @@
 package manga;
 
+import sales.Sale;
+import ui.PrintColumn;
+
 import java.util.ArrayList;
 
 //@@author xenthm
 /**
  * Represents a list of <code>Manga</code>. Extends {@code ArrayList<Manga>}
  */
 public class MangaList extends ArrayList<Manga> {
-    /**
-     * Prints the entire <code>MangaList</code> with the associated <code>deadline</code> after formatting.
-     */
-    public void print() {
-        assert size() >= 0 : "MangaList.size() must be non-negative";
+    public static ArrayList<PrintColumn<Manga>> mangaColumnsToPrint(boolean includeDeadline, boolean includeSales) {
+        ArrayList<PrintColumn<Manga>> columns = new ArrayList<>();
+
+        columns.add(new PrintColumn<>("no.", 3, null));
+        columns.add(new PrintColumn<>("Manga Name", 40, Manga::getMangaName));
 
-        for (int i = 0; i < size(); i++) {
-            System.out.println((i + 1) + ". " + get(i).getMangaName() + " | Deadline: " + get(i).getDeadline());
+        if (includeDeadline) {
+            columns.add(new PrintColumn<>("Deadline", 15, Manga::getDeadline));
         }
+
+        if (includeSales) {
+            columns.add(new PrintColumn<>("Unit Price", 10, manga -> {
+                Sale salesData = manga.getSalesData();
+                return salesData != null ? String.valueOf(salesData.getUnitPrice()) : "N/A";
+            }));
+            columns.add(new PrintColumn<>("Units Sold", 10, manga -> {
+                Sale salesData = manga.getSalesData();
+                return salesData != null ? String.valueOf(salesData.getQuantitySold()) : "N/A";
+            }));
+            columns.add(new PrintColumn<>("Revenue", 10, manga -> {
```

Comment:
What do the magic numbers `40`, `3`, `15` in these lines mean?

Reply:
They are pre-set widths for the columns.